### DISCUSSION TITLE
How to setup Next.js app, Nextra docs, and Nextra blog?

### DISCUSSION THREAD

**Discussion by justin-harr** [shuding/nextra#681] [Q&A]
I would like to setup my main application (SaaS) using Next.js and build the docs and blog for it using Nextra. I'm hoping for advice/examples on the best way to accomplish this. I see three options:

1. Create `/docs` and `/blog` directory inside my main application and install Nextra globally. This doesn't really work because the themes are applied globally and overlap/interfere with other parts of the application.

2. Setup 3 separate applications inside a [monorepo like this ](https://github.com/vercel/next.js/tree/canary/examples/with-zones). This should work because it separates the concerns/styling of each of the applications. However, the rewrites to the blog/docs don't seem to work, it either shows a blank page or broken styles.

3. Setup the blog and docs on subdomains. I haven't tried this one but don't see why it wouldn't work since they're completely different sites. However, afaik you shouldn't have your docs and blog on subdomains for SEO purposes.

Any advice on how to accomplish this?

**Comment by shuding** [comment]
1 should work. The latest beta doesn't apply the theme globally anymore (example: https://github.com/shuding/nextra/tree/core/examples/swr-site). You can install the docs theme and make /blog something like this: https://github.com/vercel/turborepo/blob/main/docs/pages/blog.mdx

**Reply by huykon** [reply]
I don't see any docs/guide how to integrate both nextra docs & blog into a nextjs project
